geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala

```diff
@@ -147,6 +155,26 @@ class KafkaDataStore(
     consumer
   }
 
+  override def getSchema(typeName: String): SimpleFeatureType = {
+    layerViewLookup.get(typeName) match {
+      case None => super.getSchema(typeName)
+      case Some(orig) =>
+        val parent = super.getSchema(orig)
+        if (parent == null) { null } else {
+          val view = config.layerViewsConfig.get(orig).flatMap(_.find(_.typeName == typeName)).getOrElse {
+            // this should be impossible since we created the lookup from the view config
+            throw new IllegalStateException("Inconsistent layer view config")
+          }
+          KafkaDataStore.createLayerView(parent, view).viewSft
+        }
+    }
+  }
+
+  override def getTypeNames: Array[String] = {
+    val nonViews = super.getTypeNames
+    nonViews ++ layerViewLookup.collect { case (k, v) if nonViews.contains(v) => k }
```

Comment:
I wonder if it'd be "nice" to log a warning when the base store does *not* have the layer configured for a view.  

E.g., the case when `!nonViews.contains(v)`.

Reply:
i think the default case would be to not have any views, so i don't think we'd want to log a warning on that, unless i'm misunderstanding you

Reply:
If `nonViews` had one SFT named `foo`, and someone computed a view on `bar`.  What happens?

I think that's the case where `nonViews.contains(v)` is false, right?

Reply:
right, yeah if you configure a view on a feature type that doesn't exist yet. I see your point now.